### DISCUSSION TITLE
Get inventory_item_id from Product Module

### DISCUSSION THREAD

**Discussion by Deroswent** [medusajs/medusa#9207] [Q&A]
In Medusa V1 you can use [ProductVariantInventoryService](https://docs.medusajs.com/references/services/classes/services.ProductVariantInventoryService)  to get  inventory_item_id  from product variant_id.   Like send product variant_id -> receive inventory_item_id associated with this product variant.

In Medusa v2 I can't find any methods to get inventory_item_id based on variant_id or product_id.  It seems that such methods just doesn't exist.  

How do I associate variant_id or product_id with inventory module and stock module?  There are no methods to get any data about inventory or stock in Product Module.

**Comment by SalahAdDin** [comment]
Wait a minute, aren't those modules linked?

It is the main idea behind all the workflow!!!

**Reply by Deroswent** [reply]
You can't use Link because the two modules are already linked in the medusa core in [LinkProductVariantInventoryItem](https://docs.medusajs.com/v2/resources/commerce-modules/product/relations-to-other-modules#inventory-module).
You will just get the error:  
`error: Unable to merge GraphQL type “ProductVariant”: The field “inventory_items” is already defined with a different type. It is declared as “LinkProductVariantInventoryItem”, but you tried to override it with “InventoryItem”`.

This is the main problem: you can't link modules because they are already linked. But you can't use variant_id -> inventory_item_id, because there is simply no method for that in any of the modules.  Although there is such a method in v1. It looks like the developers just forgot about it.

I tried using a relation, I thought if these modules have a link- it might work, but that didn't work either.


If you have any ideas on how this can be done - please share.

**Comment by carlos-r-l-rodrigues** [comment]
You can use the `query` tool to get them. This is how you aggregate data from multiple modules.

```typescript
await query.graph({
  entity: "product",
  fields: ["variants.inventory.*"],
})
```

**Reply by Deroswent** [reply]
Thank you for your response. 

It looks very strange to use Query, considering that these two modules have a link in the core Medusa [LinkProductVariantInventoryItem ](https://docs.medusajs.com/v2/resources/commerce-modules/product/relations-to-other-modules#inventory-module).

I haven't found a better option yet, it seems the developers just forgot to create some methods and now it's impossible to do it (I mean using  Medusa methods). 


I hope when the developers release an official guide on migration from v1 to v2, it will describe how to use some analog of [ProductVariantInventoryService ](https://docs.medusajs.com/references/services/classes/services.ProductVariantInventoryService ) from v1 in the new version. This is quite an important method, they can't just remove it without explaining how to link this data together.

**Reply by carlos-r-l-rodrigues** [reply]
The `query` is the recommended approach to fetch data from multiple modules. Medusa 2.0 is modular, that means a module can be even an external 3rd party service, we can't just join tables as in v1.
But obviouly we can't prevent someone to just write a plain sql query joining everything because they know it just uses a single db instance. We just don't document nor recommend this approach
